opendebates/tests/test_views.py

```diff
@@ -17,6 +17,43 @@
 reverse = partial(old_reverse, urlconf='opendebates.urls')
 
 
+class RootRedirectTest(TestCase):
+
+    def setUp(self):
+        self.site = SiteFactory()
+        # really old debate
+        DebateFactory(
+            site=self.site,
+            debate_time=timezone.now() - datetime.timedelta(days=200),
+        )
+        # semi-recent debate
+        self.previous_debate = DebateFactory(
+            site=self.site,
+            debate_time=timezone.now() - datetime.timedelta(days=20),
+        )
+        # upcoming debate
+        self.next_debate = DebateFactory(
+            site=self.site,
+            debate_time=timezone.now() + datetime.timedelta(days=1),
+        )
+
+    def tearDown(self):
+        Site.objects.clear_cache()
+
+    def test_next_debate(self):
+        DebateFactory(
+            site=self.site,
+            debate_time=timezone.now() + datetime.timedelta(days=20),
+        )
+        r = self.client.get('/')
+        self.assertRedirects(r, '/%s/' % self.next_debate.prefix)
+
+    def test_only_previous_debate(self):
+        self.next_debate.delete()
+        r = self.client.get('/')
+        self.assertRedirects(r, '/%s/' % self.previous_debate.prefix)
+
```

Comment:
Could you also add a test for the case that there is no `Debate` for the relevant `Site`?

Reply:
Good idea! Added.